CMakeLists.txt

```diff
@@ -214,7 +214,9 @@ function(FIND_BUNDLED_PYTHON)
     	set(Python_INCLUDE_DIRS "${THIRD_PARTY_DIR}/Python${PYTHONVERSION}/include")
     else()
         set(Python_SITELIB "${PYTHONPATH}/site-packages")
-        set(Python_INCLUDE_DIRS "${THIRD_PARTY_INC_DIR}/python${PYTHONVERSION}")
+        if (NOT DEFINED Python_INCLUDE_DIRS)
+            set(Python_INCLUDE_DIRS "${THIRD_PARTY_INC_DIR}/python${PYTHONVERSION}")
```

Comment:
CMake provides `option` for user-editable parameters (e.g. by passing in with `-D`)

Reply:
That's right, but cmake will not override variables that are already defined.  For example if we do `cmake -DTHRID_PARTY_DIR=/usr/local`, the third party directory will remain unchanged since it's defined in CMakeLists.

Here, if `Python_INCLUDE_DIRS` is not defined by the user we use the default value.  For the CentOS7 and Ubuntu18 builds, we need to define these values with the command line.

Reply:
Take a look at the `BUILD_GUI` parameter which uses `option`.

Reply:
That does look like another cleaner way to do it, and a bit cleaner too.

Would you be ok with keeping the current technique, since it will be removed after we release 3.9?  These rules only apply to CentOS7 which will be deprecated after 3.9, and Ubuntu18 which will be removed from our test suite after #3401 is in place.  If not I can go ahead and use cmake's `option` command and go through another round of testing.

Reply:
Fine with me